rta/evasion_patch_etw_amsi.py

```diff
@@ -0,0 +1,85 @@
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License
+# 2.0; you may not use this file except in compliance with the Elastic License
+# 2.0.
+
+from . import common
+from . import RtaMetadata
+from ctypes import windll
+from ctypes import wintypes
+import ctypes
+import platform, time
+
+kernel32 = windll.kernel32
+
+
+LoadLibraryA = kernel32.LoadLibraryA
+LoadLibraryA.argtypes = [wintypes.LPCSTR]
+LoadLibraryA.restype = wintypes.HMODULE
+
+GetProcAddress = kernel32.GetProcAddress
+GetProcAddress.argtypes = [wintypes.HMODULE, wintypes.LPCSTR]
+GetProcAddress.restype = ctypes.c_void_p
+
+VirtualProtect = kernel32.VirtualProtect
+VirtualProtect.argtypes = [wintypes.LPVOID, ctypes.c_size_t, wintypes.DWORD, wintypes.PDWORD]
+VirtualProtect.restype = wintypes.BOOL
+
+GetCurrentProcess = kernel32.GetCurrentProcess
+GetCurrentProcess.restype = wintypes.HANDLE
+
+WriteProcessMemory = kernel32.WriteProcessMemory
+WriteProcessMemory.argtypes = [wintypes.HANDLE, wintypes.LPVOID, wintypes.LPCVOID, ctypes.c_size_t, wintypes.LPVOID]
+WriteProcessMemory.restype = wintypes.BOOL
+
+GetModuleHandleA = kernel32.GetModuleHandleA
+GetModuleHandleA.restype = wintypes.HANDLE
+GetModuleHandleA.argtypes = [ wintypes.LPCSTR ]
+
+
+RWX = 0x40 # PAGE_READ_WRITE_EXECUTE
+OLD_PROTECTION = wintypes.LPDWORD(ctypes.c_ulong(0))
+
+metadata = RtaMetadata(
+    uuid="395d0e4c-e7f5-4c77-add7-92b1d2ba169e",
+    platforms=["windows"],
+    endpoint=[],
+    siem=[{"rule_id": "586bf106-b208-45fc-9401-727664175ca0", "rule_name": "Potential AMSI Bypass via Memory Patching"},
+          {"rule_id": "3046168a-91cb-4ecd-a061-b75b1df1c107", "rule_name": "Potential Evasion via Event Tracing Patching"}],
+    techniques=["T1562.001"],
+)
+
+
+@common.requires_os(metadata.platforms)
+
+def main():
+    if platform.architecture()[0] == '64bit':
+        print('[+] using x64 based patch')
+        patch = (ctypes.c_char * 6)(0x90, 0x90, 0x90, 0x90, 0x90, 0x90)
+    if platform.architecture()[0] != '64bit':
+        print('[+] using x86 based patch')
+        patch = (ctypes.c_char * 8)(0x90, 0x90, 0x90, 0x90, 0x90, 0x90, 0x90, 0x90)
```

Comment:
Q: These are non-destructive patches right?

Reply:
yes 0x90 is translated to nop instruction (no operation, dummy patch and process will exist immediately after patch/rta is done)

![image](https://github.com/elastic/detection-rules/assets/64742097/4cffd769-443f-4346-a967-f88b8e6ea971)